### ISSUE TITLE
.NET Release - 5.0 SDK

### ISSUE THREAD

**Issue by mthalman** [dotnet/dotnet-docker#2830]
# .NET Release

## Release Versions

_The set of .NET versions that are being released as a unit._

* 5.0.300

## 1. Main Branch Tasks

1. - [x] Merge appropriate commits from nightly branch.  Things to check:
      - [ ] New/EOL distro
      - [ ] Infrastructure changes
      - [ ] PowerShell version
      - [ ] Check for additional changes by diffing the main and nightly branches
      - [ ] &lt;add link to PR/commit&gt;
1. - [x] Check whether a change has been made to a Dockerfile that is shared by multiple .NET versions. If a change has been made and the .NET versions that share that file are not being released at the same time, define a separate Dockerfile to isolate the change to the .NET version that's being released. Conversely, after a shared Dockerfile has diverged in such a way, it should be combined again into a shared Dockerfile when the other other .NET version is released. Shared Dockerfiles to check:
      - [ ] 3.1 runtime-deps shared with 5.0 and 6.0
1. - [x] Wait for .NET archive files (.zip, .tar.gz) to be available at blob storage location
1. - [x] Run `update-dependencies` tool to update all the necessary files to reflect the specified .NET versions (run this command for each version being released):
      - [ ] `dotnet run --project .\eng\update-dependencies\update-dependencies.csproj 5.0 --product-version sdk=5.0.300`
1. - [x] Inspect generated changes for correctness
1. - [x] Commit generated changes
1. - [x] Create PR
1. - [x] Get PR signoff
1. - [x] Merge PR
1. - [x] Wait for changes to be mirrored to internal [dotnet-docker repo](https://dev.azure.com/dnceng/internal/_git/dotnet-dotnet-docker) (internal MSFT link)
1. - [x] Build images - Queue build stage of [dotnet-docker pipeline](https://dev.azure.com/dnceng/internal/_build?definitionId=373) (internal MSFT link) with variables:

      All releases:

          stages: build

      Servicing release (exclude 5.0/6.0 if separate runtime-deps Dockerfiles were defined):

          noCache: true
          imageBuilder.pathArgs: --path 'src/*/3.1/*' --path 'src/*/5.0/*' --path 'src/*/6.0/*' --path 'src/monitor/*'

1. - [x] Wait for NuGet packages to be published during release tic-toc
1. - [x] Test and publish images - Queue build of [dotnet-docker pipeline](https://dev.azure.com/dnceng/internal/_build?definitionId=373) (internal MSFT link) with variables:

      All releases:

          stages: test;publish
          sourceBuildId: <Build ID from the build stage>

      Servicing release (exclude 5.0/6.0 if separate runtime-deps Dockerfiles were defined):

          imageBuilder.pathArgs: --path 'src/*/3.1/*' --path 'src/*/5.0/*' --path 'src/*/6.0/*' --path 'src/monitor/*'

1. - [x] Confirm READMEs have been updated in [Docker Hub](https://hub.docker.com/_/microsoft-dotnet)

## 2. Nightly Branch Tasks (Not needed for Preview-only release)

1. - [x] Merge main branch to nightly
1. - [x] Run the same `update-dependencies` command(s) in nightly that were run for the main branch
1. - [x] Inspect generated changes for correctness
1. - [x] Commit generated changes
1. - [x] Create PR
1. - [x] Get PR signoff
1. - [x] Merge PR
1. - [x] Wait for automatically queued CI build to finish on [dotnet-docker-nightly pipeline](https://dev.azure.com/dnceng/internal/_build?definitionId=359) (internal MSFT link)
1. - [x] Confirm READMEs have been updated in [Docker Hub](https://hub.docker.com/_/microsoft-dotnet)

**Comment by Gladskih** [comment]
And you have broken PAT authentication for AzureDevOps nuget feeds.

**Comment by KoenWillemse** [comment]
We're running into issue with the 5.0-focal docker image. All our builds used to work fine until the update of yesterday. We're running into issues with packing. We get following error now:
```
/usr/share/dotnet/sdk/5.0.300/Sdks/NuGet.Build.Tasks.Pack/build/NuGet.Build.Tasks.Pack.targets(221,5): error NU5050: Attempted to pack multiple files into the same location(s). The following destinations were used multiple times: ./api-ms-win-core-util-l1-1-0.dll, ./Microsoft.AspNetCore.DataProtection.Extensions.dll, ./api-ms-win-crt-math-l1-1-0.dll,
```

We were not experiencing any issues until today.

**Comment by melihc1** [comment]
Can you provide a stable tag for images based on latest stable release versions?

`https://github.com/dotnet/sdk/releases`

**Comment by mthalman** [comment]
While we investigate, you can use the 5.0.203 tag, which was the previous version, to become unblocked. You can see the list of current tags at https://hub.docker.com/_/microsoft-dotnet-sdk. By replacing `5.0.300` with `5.0.203`, it'll give you the appropriate previous version.

**Comment by mthalman** [comment]
> We're running into issue with the 5.0-focal docker image. All our builds used to work fine until the update of yesterday. We're running into issues with packing. We get following error now:
> 
> ```
> /usr/share/dotnet/sdk/5.0.300/Sdks/NuGet.Build.Tasks.Pack/build/NuGet.Build.Tasks.Pack.targets(221,5): error NU5050: Attempted to pack multiple files into the same location(s). The following destinations were used multiple times: ./api-ms-win-core-util-l1-1-0.dll, ./Microsoft.AspNetCore.DataProtection.Extensions.dll, ./api-ms-win-crt-math-l1-1-0.dll,
> ```
> 
> We were not experiencing any issues until today.

@zkat - Can you comment on this?  It looks related to https://github.com/NuGet/NuGet.Client/pull/3923.  Was this change introduced in .NET 5.0.300 SDK?

**Comment by mthalman** [comment]
> And you have broken PAT authentication for AzureDevOps nuget feeds.
> It was ok before like this:
> `RUN dotnet nuget add source ${feed_uri} --name ${feed_name} --username "whatever" --password=${PAT} --store-password-in-clear-text --valid-authentication-types "basic"`
> And now restore step get 403 Forbidden.

@Gladskih - Unfortunately, I can't repro this.  I tried with both Linux and Windows containers on 5.0.300 SDK and it worked when targeting a private Azure DevOps feed.  Have you verified that reverting back to the previous version of the SDK resolves your issue (see https://github.com/dotnet/dotnet-docker/issues/2830#issuecomment-848749790)?  Please feel free to log a separate issue and provide more detail such as your Docker environment, Dockerfile, project file, etc.

**Comment by Gladskih** [comment]
@mthalman Yes, I used exactly reverting tonight to fix the build.

**Comment by zkat** [comment]
Yes, we made a decision to let this change in: https://github.com/NuGet/Home/issues/6941#issuecomment-784568222

If you're running into this issue, you've been non-deterministically clobbering files during pack this whole time, which we consider to have been a bug.

**Comment by MichaelSimons** [comment]
[Triage] Closing as the release is complete.  If you have encountered any issues with the 5.0.300 sdk images, please open new issues.
